spec/controllers/diaspora_federation/webfinger_controller_spec.rb

```diff
@@ -35,31 +35,77 @@ module DiasporaFederation
       end
     end
 
-    describe "GET #legacy_webfinger", rails: 5 do
-      it "succeeds when the person exists" do
-        get :legacy_webfinger, params: {q: alice.diaspora_id}
+    describe "GET #webfinger", rails: 5 do
+      it "uses the JRD format as default" do
+        get :webfinger, params: {resource: alice.diaspora_id}
         expect(response).to be_success
+        expect(response.header["Content-Type"]).to include "application/jrd+json"
       end
 
-      it "succeeds with 'acct:' in the query when the person exists" do
-        get :legacy_webfinger, params: {q: "acct:#{alice.diaspora_id}"}
-        expect(response).to be_success
-      end
+      context "json" do
+        it "succeeds when the person exists" do
+          get :webfinger, format: :json, params: {resource: alice.diaspora_id}
+          expect(response).to be_success
+        end
 
-      it "contains the diaspora* ID" do
-        get :legacy_webfinger, params: {q: "acct:#{alice.diaspora_id}"}
-        expect(response.body).to include "<Subject>acct:alice@localhost:3000</Subject>"
-      end
+        it "succeeds with 'acct:' in the query when the person exists" do
+          get :webfinger, format: :json, params: {resource: "acct:#{alice.diaspora_id}"}
+          expect(response).to be_success
+        end
+
+        it "contains the diaspora* ID" do
+          get :webfinger, format: :json, params: {resource: "acct:#{alice.diaspora_id}"}
+          expect(response.body).to include "\"subject\": \"acct:alice@localhost:3000\""
+        end
+
+        it "returns a application/jrd+json" do
+          get :webfinger, format: :json, params: {resource: "acct:#{alice.diaspora_id}"}
+          expect(response.header["Content-Type"]).to include "application/jrd+json"
+        end
+
+        it "404s when the person does not exist" do
+          get :webfinger, format: :json, params: {resource: "me@mydiaspora.pod.com"}
+          expect(response).to be_not_found
+        end
 
```

Comment:
RFC WebFinger requires that if we miss the `resource` parameter we fail with `400` status. It is how we work now, but since this is a specification, maybe we should add an example for this case as well?